### DISCUSSION TITLE
Variable Data Reference from Examples

### DISCUSSION THREAD

**Discussion by joseym** [postmanlabs/postman-flows#260] [Defect]
It wasn't immediately clear to me, but request blocks take the variable reference form the response format of an example on the request, pretty slick.

However, I like to create examples for successful requests and various expected errors. 

Sadly the variable reference in the **Send Request** doesn't allow me to define which example it should be using and my flows are  seemingly taking random error examples and throwing reference errors...

1. Can we add a way to define which example it uses
2. How can I currently guarantee the success example is used for variable reference?

**Comment by saswatds** [comment]
Hey, @joseym, Ideally flows should pick up the examples with 2xx status code as the reference. Let me take a look if there is some issue here.

**Reply by mikedfunk** [reply]
That doesn't seem ideal to me. What if you want to create a flow that uses 201, but you also have a 200 response defined? Or if you have a legitimate flow that depends on a 4xx response?

**Reply by hayden-plentific** [reply]
I'm having the same experience as @joseym, seems to ignore the status code and picks up the data structure of the last saved example, would be only slightly annoying if this was applied exclusively at build time, but because it's applied dynamically, the creation of a new example can invalidate all of my existing flows.

I think it would make sense if it adopted a more liberal approach & took the superset of all fields included across your set of saved examples.

**Comment by jasperroel** [comment]
I was also playing around with this. In my case, it helped to drag the example I wanted to work to the bottom of the list in Postman (so, after the error conditions). This way, it seemed to pick up that example so I could work with the 2xx example.

But I agree it would be nice to specify which example to work with.
Or as an alternative:
- Have Postman flow offer multiple suggestion (basically, allow us to pick the model ad-hoc in the block
- Allow the user to specify the example path (`/variables/body/id` for example). If I try that current, Postman will unhelpfully surround that with quotes and then send that string verbatim to the endpoint.

**Comment by saswatds** [comment]
Hey @mikedfunk @hayden-plentific @jasperroel  thanks a lot for the feedback.

I have fixed a bunch of these issues with the type inference from examples for different status codes. It should be much more liberal in the upcoming releases.

The behaviour I have implemented is that If you have various status codes like `200`, `201` or say `400`. The resultant auto-suggestion will show a combination of everything. Additionally, I have also relaxed the expression input to let you enter any path that you deem fit.

The only thing still left to implement is what happens when there are multiple responses for the same status code. I will keep working to improve that. Inference functions are always a bit tricky to implement for polymorphic responses. 

We are working on a lot of improvements so stay tuned. Thanks.

**Reply by hayden-plentific** [reply]
That's great. Looking forward to seeing the release, thanks @saswatds!
